yt/frontends/swift/data_structures.py

```diff
@@ -0,0 +1,209 @@
+
+        # Not used in this function, but passed to parameters
+        hydro = self._get_info_attributes("HydroScheme")
+        subgrid = self._get_info_attributes("SubgridScheme")
+
+        self.domain_right_edge = header["BoxSize"]
+        self.domain_left_edge = np.zeros_like(self.domain_right_edge)
+
+        self.dimensionality = int(header["Dimension"])
+
+        # SWIFT is either all periodic, or not periodic at all
+        periodic = int(runtime_parameters["PeriodicBoundariesOn"])
+
+        if periodic:
+            self.periodicity = [True] * self.dimensionality
+        else:
+            self.periodicity = [False] * self.dimensionality
+
+        self.current_time = float(header["Time"])
+
+        # Now cosmology enters the fray, as a runtime parameter.
+        self.cosmological_simulation = int(policy["cosmological integration"])
+
+        if self.cosmological_simulation:
+            try:
+                self.current_redshift = float(header["Redshift"])
+                # These won't be present if self.cosmological_simulation is false
+                self.omega_lambda = float(parameters["Cosmology:Omega_lambda"])
+                self.omega_matter = float(parameters["Cosmology:Omega_m"])
+                # This is "little h"
+                self.hubble_constant = float(parameters["Cosmology:h"])
+            except KeyError:
+                mylog.info(
+                    "Could not find cosmology information in Parameters."
+                )
+                mylog.info(
+                    "Setting up as a non-cosmological run. Check this!"
+                )
+                self.cosmological_simulation = 0
+                self.current_redshift = 0.0
+                self.omega_lambda = 0.0
+                self.omega_matter = 0.0
+                self.hubble_constant = 0.0
+        else:
+            self.current_redshift = 0.0
+            self.omega_lambda = 0.0
+            self.omega_matter = 0.0
+            self.hubble_constant = 0.0
+
+
+        # Store the un-parsed information should people want it.
+        self.parameters = dict(
+            header=header,
+            runtime_parameters=runtime_parameters,
+            policy=policy,
+            parameters=parameters,
+            hydro=hydro,
+            subgrid=subgrid)
+
+        ## TODO: we need to deal with num_files per snapshot
+        self.file_count = 1
+        self.filename_template = self.parameter_filename
+
+        return
+
+    @classmethod
+    def _is_valid(self, *args, **kwargs):
+        """
+        Checks to see if the file is a valid output from SWIFT.
+        This requires the file to have the Code attribute set in the
+        Header dataset to "SWIFT".
+        """
+        filename = args[0]
+        valid = True
+        # Attempt to open the file, if it's not a hdf5 then this will fail:
+        try:
+            handle = h5py.File(filename, "r")
+        except:
```

Comment:
never use bare except: https://stackoverflow.com/questions/14797375/should-i-always-specify-an-exception-type-in-except-statements

I know there are places in other frontends where we do that. We should fix those frontends.

Reply:
You are looking for `except (FileNotFoundError, OSError) as e` as far as I am aware.